### DISCUSSION TITLE
onProgressChanged appears only when Loading in iOS, but not when changing pages.

### DISCUSSION THREAD

**Discussion by SangHoon-Han-Dev** [pichillilorenzo/flutter_inappwebview#2422] [Q&A]
Hello.
onProgressChanged appears only when Loading in iOS, but not when changing pages.
In Android, it is called not only when Loading, but also when changing pages.

`initialSettings: InAppWebViewSettings(
              javaScriptEnabled: true,
              mediaPlaybackRequiresUserGesture: false,
              allowsInlineMediaPlayback: true,
              useHybridComposition: true,
            ),`

`onProgressChanged: (controller, progress) {
              print("WebViewPage onProgressChanged: $progress%");
            },`

flutter_inappwebview: ^6.1.0+1
iOS: 17.3.1
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.7.1 23H222 darwin-arm64, locale ko-KR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)

Is there anything else I need to set?
Thank you.

**Comment by pichillilorenzo** [comment]
No, it should work out of the box as it is a very basic event.
Please, post a code to reproduce the issue and post the logs that you get when loading the page and changing the page, thanks.

**Reply by SangHoon-Han-Dev** [reply]
I added my source and log.

Navigate the screen with javaScript on the Web.

**Comment by SangHoon-Han-Dev** [comment]
Thank you.

My source code and Android, iOS logs are below.
I'm new to using the question function, so I still don't know how to make it look good.
Sorry.

**[Source Code]**

import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';
import 'package:get/get.dart';
import 'package:hush/controllers/main_controller.dart';

import '../common/Const.dart';
import '../controllers/dialog_controller.dart';

class WebViewPageTest extends StatefulWidget {
@override
_WebViewPageTestState createState() => _WebViewPageTestState();
}

class _WebViewPageTestState extends State<WebViewPageTest>
with WidgetsBindingObserver {
InAppWebViewController? webViewController;

@override
void initState() {
super.initState();
WidgetsBinding.instance.addObserver(this);
}

@override
void dispose() {
WidgetsBinding.instance.removeObserver(this);
super.dispose();
}

void showExitDialog() {
Get.defaultDialog(
title: "",
content: const Text(
"앱을 종료하시겠습니까?",
style: TextStyle(fontSize: 18, color: Colors.black),
),
confirm: ElevatedButton(
onPressed: () {
SystemNavigator.pop(); // 앱 종료
},
child: const Text(
"확인",
style: TextStyle(fontSize: 16, color: Colors.white),
),
style: ElevatedButton.styleFrom(
backgroundColor: Colors.black,
shape: RoundedRectangleBorder(
borderRadius: BorderRadius.circular(5),
),
),
),
cancel: ElevatedButton(
onPressed: () {
Get.back();
},
style: ElevatedButton.styleFrom(
backgroundColor: Colors.white,
shape: RoundedRectangleBorder(
borderRadius: BorderRadius.circular(5),
),
),
child: const Text(
"취소",
style: TextStyle(fontSize: 16, color: Colors.black),
),
),
);
}

Future<String> getCurrentUrl() async {
// Uri? currentUri = await webViewController?.getUrl();
Uri? currentUri = await getCurrentUri();
String currentUrl = currentUri?.toString() ?? "URL을 가져올 수 없음";
print('[WebViewPageTest] getCurrentUrl currentUrl : $currentUrl');

return currentUrl;
}

Future<Uri> getCurrentUri() async {
Uri? currentUri = await webViewController!.getUrl();
print('[WebViewPageTest] getCurrentUri currentUri : $currentUri');

return currentUri!;
}

String mainPageUrl = '${MainController.to.domainUrl.value}$URL_MAIN';

@override
Widget build(BuildContext context) {
return SafeArea(
child: PopScope(
canPop: false,
onPopInvoked: (bool didPop) async {
String currentUrl = await getCurrentUrl();

if (webViewController != null) {
if (currentUrl == '${MainController.to.domainUrl.value}$URL_MAIN' ||
currentUrl ==
'${MainController.to.domainUrl.value}$URL_SUB_REGISTER') {
showExitDialog();
return;
}

bool canGoBack = await webViewController!.canGoBack();

if (canGoBack) {
webViewController!.goBack();
return;
} else {
print('[WebViewPageTest] build PopScope onPopInvoked 이전 페이지 없음');
showExitDialog();
}
}
},
child: Scaffold(
body: InAppWebView(
initialUrlRequest:
URLRequest(url: WebUri(MainController.to.domainUrl.value)),
initialSettings: InAppWebViewSettings(
javaScriptEnabled: true,
mediaPlaybackRequiresUserGesture: false,
supportZoom: false,
allowsInlineMediaPlayback: true,
useHybridComposition: true,
),
onWebViewCreated: (controller) {
print("[WebViewPageTest] onWebViewCreated");
webViewController = controller;
},
onLoadStart: (controller, url) {
print("[WebViewPageTest] onLoadStart: $url");
DialogController.to.updateLoadingState(true);
},
onLoadStop: (controller, url) {
print("[WebViewPageTest] onLoadStop: $url");
DialogController.to.updateLoadingState(false);
},
onLoadError: (controller, url, code, message) {
print("[WebViewPageTest] onLoadError: $message");
},
onProgressChanged: (controller, progress) {
print("[WebViewPageTest] onProgressChanged: $progress%");
if (progress == 100) {
DialogController.to
.updateLoadingState(false, progress: progress);
} else if (progress != 100 &&
!DialogController.to.isLoading.value) {
DialogController.to
.updateLoadingState(true, progress: progress);
}
},
onConsoleMessage: (controller, consoleMessage) {
print(
"[WebViewPageTest] onConsoleMessage: ${consoleMessage.message}");
},
onPageCommitVisible: (controller, url) {
print("[WebViewPageTest] onPageCommitVisible: $url");
},
onTitleChanged: (controller, title) {
print("[WebViewPageTest] onTitleChanged: $title");
},
onEnterFullscreen: (controller) {
print("[WebViewPageTest] onEnterFullscreen");
},
onExitFullscreen: (controller) {
print("[WebViewPageTest] onExitFullscreen");
},
onPermissionRequest: (controller, request) async {
print("[WebViewPageTest] onPermissionRequest");
print(
"[WebViewPageTest] onPermissionRequest request.resources: ${request.resources}");
return PermissionResponse(
resources: request.resources,
action: PermissionResponseAction.GRANT);
},
),
),
),
);
}
}



**Android ========================================================================================
[Load log]**
I/flutter (28846): [WebViewPageTest] onLoadStart: https://hsh.test.net/
I/flutter (28846): [MainController] _updateLoadingState state: true, progress: 0
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onLoadStart" using {url: https://hsh.test.net/}
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onUpdateVisitedHistory" using {isReload: false, url: https://hsh.test.net/}
I/flutter (28846): [WebViewPageTest] onProgressChanged: 70%
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onProgressChanged" using {progress: 70}
I/flutter (28846): [WebViewPageTest] onTitleChanged: Hush
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onTitleChanged" using {title: Hush}
D/InputTransport(28846): Input channel destroyed: 'ClientS', fd=276
D/InputTransport(28846): Input channel destroyed: 'ClientS', fd=277
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onUpdateVisitedHistory" using {isReload: false, url: https://hsh.test.net/}
I/flutter (28846): [WebViewPageTest] onLoadStop: https://hsh.test.net/
I/flutter (28846): [MainController] _updateLoadingState state: false, progress: 0
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onLoadStop" using {url: https://hsh.test.net/}

**[Page Move log]**
I/ViewRootImpl@85146c9[MainActivity](28846): ViewPostIme pointer 0
D/CompatibilityChangeReporter(28846): Compat change id reported: 289878283; UID 10633; state: ENABLED
I/GestureDetector(28846): obtain mCurrentDownEvent. id: 926382379 caller: com.pichillilorenzo.flutter_inappwebview_android.webview.in_app_webview.InAppWebView$5.onTouch:516 android.view.View.dispatchTouchEvent:16230 android.view.ViewGroup.dispatchTransformedTouchEvent:3346 
D/InputMethodManagerUtils(28846): startInputInner - Id : 0
I/InputMethodManager(28846): startInputInner - IInputMethodManagerGlobalInvoker.startInputOrWindowGainedFocus
I/GestureDetector(28846): obtain mCurrentMotionEventRaw. action: 2 id: 166432652
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onWindowFocus" using []
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onWindowFocus" using []
I/ViewRootImpl@85146c9[MainActivity](28846): ViewPostIme pointer 1
I/ViewRootImpl@85146c9[MainActivity](28846): onDisplayChanged oldDisplayState=2 newDisplayState=2
I/flutter (28846): [WebViewPageTest] onProgressChanged: 10%
I/flutter (28846): [MainController] _updateLoadingState state: true, progress: 10
I/flutter (28846): [WebViewPageTest] onProgressChanged: 100%
I/flutter (28846): [MainController] _updateLoadingState state: false, progress: 100
I/flutter (28846): [WebViewPageTest] onProgressChanged: 100%
I/flutter (28846): [MainController] _updateLoadingState state: false, progress: 100
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onReceivedTouchIconUrl" using {precomposed: false, url: https://hsh.test.net/logo192.png}
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onProgressChanged" using {progress: 10}
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onUpdateVisitedHistory" using {isReload: false, url: https://hsh.test.net/metaverse/dating/register/phone}
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onProgressChanged" using {progress: 100}
[AndroidInAppWebViewController] (android) WebView ID 0 calling "onProgressChanged" using {progress: 100}`

**iOS ========================================================================================**
**[Load log]**
flutter: [WebViewPageTest] onLoadStart: https://hsh.test.net/
flutter: [MainController] _updateLoadingState state: true, progress: 0
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onLoadStart" using {url: https://hsh.test.net/}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onZoomScaleChanged" using {oldScale: 1.0, newScale: 0.38269031047821045}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onContentSizeChanged" using {oldContentSize: {height: 0.0, width: 0.0}, newContentSize: {width: 375.0, height: 734.0}}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onContentSizeChanged" using {oldContentSize: {height: 1918.0, width: 980.0}, newContentSize: {width: 375.0, height: 734.0}}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onReceivedServerTrustAuthRequest" using {protectionSpace: {distinguishedNames: null, protocol: https, realm: null, proxyType: null, sslError: {code: 4, message: Indicates the evaluation succeeded and the certificate is implicitly trusted, but user intent was not explicitly specified.}, port: 443, sslCertificate: {x509Certificate: [48, 130, 5, 225, 48, 130, 4, 201, 160, 3, 2, 1, 2, 2, 16, 6, 9, 212, 43, 9, 92, 84, 51, 30, 195, 179, 2, 236, 184, 173, 66, 48, 13, 6, 9, 42, 134, 72, 134, 247, 13, 1, 1, 11, 5, 0, 48, 60, 49, 11, 48, 9, 6, 3, 85, 4, 6, 19, 2, 85, 83, 49, 15, 48, 13, 6, 3, 85, 4, 10, 19, 6, 65, 109, 97, 122, 111, 110, 49, 28, 48, 26, 6, 3, 85, 4, 3, 19, 19, 65, 109, 97, 122, 111, 110, 32, 82, 83, 65, 32, 50, 48, 52, 56, 32, 77, 48, 50, 48, 30, 23, 13, 50, 52, 49, 49, 49, 56, 48, 48, 48, 48, 48, 48, 90, 23, 13, 50, 53, 49, 50, 49, 56, 50, 51, 53, 57, 53, 57, 90, 48, 27, 49, 25, 48, 23, 6, 3, 85, 4, 3, 19, 16, 105, 98, 105, 122, 115, 111, 102, 116, 1...
flutter: [WebViewPageTest] onProgressChanged: 50%
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onProgressChanged" using {progress: 50}
flutter: [WebViewPageTest] onPageCommitVisible: https://hsh.test.net/
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onPageCommitVisible" using {url: https://hsh.test.net/}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onTitleChanged" using {title: Hush}
flutter: [WebViewPageTest] onTitleChanged: Hush
flutter: [WebViewPageTest] onProgressChanged: 100%
flutter: [MainController] _updateLoadingState state: false, progress: 100
flutter: [WebViewPageTest] onLoadStop: https://hsh.test.net/
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onProgressChanged" using {progress: 100}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onLoadStop" using {url: https://hsh.test.net/}


**[Page Move log]**
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onWindowFocus" using []
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onWindowFocus" using []
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onUpdateVisitedHistory" using {isReload: null, url: https://hsh.test.net/metaverse/dating/register/phone}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onContentSizeChanged" using {newContentSize: {width: 375.0, height: 713.0}, oldContentSize: {width: 375.0, height: 734.0}}
[IOSInAppWebViewController] (iOS) WebView ID 0 calling "onContentSizeChanged" using {newContentSize: {width: 375.0, height: 574.0}, oldContentSize: {width: 375.0, height: 713.0}}
